### DISCUSSION TITLE
Applying stress boundary condition

### DISCUSSION THREAD

**Discussion by zzzmx-josh** [idaholab/moose#25320] [Q&A Modules: Solid mechanics] [answered]
Hi there,
I am working on a TensorMechanics problem using MOOSE and am trying to model the stress field from the AFM (Atom Force Microscopy) tip pressing using the Hertzian contact mechanics of a spherical indenters. The normal stress distribution on the model surface is written as
<img width="237" alt="image" src="https://github.com/idaholab/moose/assets/95216887/323bf64d-1b39-4f56-af10-1d33d763ebf8">
where r, a, and F denote the distance to the contact center, the contact radius, and the magnitude of the loading force, respectively.  As the variables in TensorMechanics are displacements, how can the stress field above be applied in the surface? 

More generally, I would like to work out how to accurately apply a stress boundary condition to solid elements in Moose. For example, the traction-free boundary condition is specified as **σ**·n = 0.  
 
Many thanks in advance

**Comment by lindsayad** [comment]
You can achieve the traction-free boundary condition by simply not specifying any boundary conditions. This yields the *natural* boundary condition. In finite elements when you integrate by parts, your surface term is <psi, sigma * n>. If you specify no boundary condition, this implicitly is setting sigma * n = 0. For your more specific boundary condition, you would likely need to write your own BC, unless such a BC already exists and I'm not aware of it. @recuero has done work with Hertzian contact

**Reply by recuero** [reply, marked as answer]
A Hertzian assumption is small deformation. You can create a function that depends on the coordinate to apply the stress as a pressure boundary condition (which is automatically normal to the boundary). If that's not good enough for your problem, you can always fall back to @lindsayad's suggestion of writing your own BC.

**Reply by zzzmx-josh** [reply]
Thank you all for your guidance and suggestions. I will give it a try.
